R/proteinToX.R

```diff
@@ -324,18 +324,18 @@ proteinToGenome <- function(x, db, id = "name", idType = "protein_id") {
     message("Checking CDS and protein sequence lengths ... ", appendLF = FALSE)
     cds_genome <- .cds_matching_protein(db, cds_genome)
     are_ok <- unlist(lapply(cds_genome, function(z) {
-        if (is(z, "GRangesList"))
+        if (is(z, "GRangesList") & length(z) > 0)
             all(z[[1]]$cds_ok)
         else NA
     }))
+    message(sum(are_ok, na.rm = TRUE), "/", length(are_ok), " OK")
     are_ok <- are_ok[!is.na(are_ok)]
     ## We've got now a list of GRanges
-    message(sum(are_ok), "/", length(are_ok), " OK")
     ## Perform the mapping for each input range with each mapped cds
     res <- mapply(
         cds_genome, as(coords_cds, "IRangesList"), as(x, "IRangesList"),
         FUN = function(gnm, cds, prt) {
-            if (is.null(gnm)) {
+            if (is.null(gnm) | !length(gnm)) {
```

Comment:
It should suffice to just have `if (!length(gnm))` here, since `lenght(NULL)` is also `0`.

Reply:
In addition, can you please change

```r
mcols(maps)$protein_start <- start(prt)
mcols(maps)$protein_end <- end(prt)
```

below to 

```r
mcols(maps) <- cbind(mcols(maps), 
                                     protein_start = start(prt),
                                     protein_end = end(prt))
```

`$` on a `DataFrame` is for some reasons slower than `cbind`.

Reply:
All addressed in PR update - well spotted that `length(gnm)` was enough!